doc/manual/src/contributing/hacking.md

```diff
@@ -99,8 +99,70 @@ You can run the whole testsuite with `make check`, or the tests for a specific c
 ### Functional tests
 
 The functional tests reside under the `tests` directory and are listed in `tests/local.mk`.
-The whole testsuite can be run with `make install && make installcheck`.
-Individual tests can be run with `make tests/{testName}.sh.test`.
+Each test is a bash script.
+
+The whole testsuite can be run with:
+```shell-session
+$ make install && make installcheck
+ran test tests/foo.sh... [PASS]
+ran test tests/bar.sh... [PASS]
+...
+```
+
+Individual tests can be run with:
+```shell-session
+$ make tests/${testName}.sh.test
+ran test tests/${testName}.sh... [PASS]
+```
+or without `make`, with:
+```shell-session
+$ ./mk/run_test.sh tests/${testName}.sh
+ran test tests/${testName}.sh... [PASS]
+```
+
+To see the output unsuppressed, one can also run
+```shell-session
+$ ./mk/debug_test.sh tests/${testName}.sh
++ foo
+output from foo
++ bar
+output from bar
+...
+```
+The test script will be traced with `set -x` and the output displayed as it happens, regardless of whether the test succeeds or fails.
```

Comment:
Add whitespace between paragraphs for better source readability, also minor rewording of instructions. Would be great if you could do that for the other parts, too.

````suggestion
The whole testsuite can be run with:

```shell-session
$ make install && make installcheck
ran test tests/foo.sh... [PASS]
ran test tests/bar.sh... [PASS]
...
```

Individual tests can be run with `make`:

```shell-session
$ make tests/${testName}.sh.test
ran test tests/${testName}.sh... [PASS]
```

or without `make`:

```shell-session
$ ./mk/run_test.sh tests/${testName}.sh
ran test tests/${testName}.sh... [PASS]
```

To see the complet eoutput, one can also run:

```shell-session
$ ./mk/debug_test.sh tests/${testName}.sh
+ foo
output from foo
+ bar
output from bar
...
```

The test script will be traced with `set -x` and the output displayed as it happens, regardless of whether the test succeeds or fails.
````